### DISCUSSION TITLE
Error connecting to your code-server instance

### DISCUSSION THREAD

**Discussion by SamHep0803** [coder/code-server#2738] [Q&A] [answered]
When running the server with `code-server --link`, upon authenticating with GitHub, the error "Error connecting to your code-server instance" is displayed. 

Any help would be appreciated!

**Comment by nooobcoder** [comment]
Attach a screenshot and explain the error you are facing in more detail. With such a short description of your problem, it is very difficult to get into the roots of your problem @SamHep0803

**Reply by MitchHamill** [reply]
@nooobcoder I'm getting this issue too, ran `code-server --link` and used [the digital ocean one click script](https://github.com/cdr/deploy-code-server/blob/main/deploy-vm/launch-code-server-linode.sh)
This is the error coming out:
<img width="1819" alt="Screen Shot 2021-04-07 at 8 42 54 pm" src="https://user-images.githubusercontent.com/57971013/113854415-3440d800-97e2-11eb-811e-66ef71e32a9a.png">
Looking forward to my issue eventually resolving itself like it did for @SamHep0803 🤨

**Reply by nooobcoder** [reply]
@MitchHamill can you attach the logs produced by code-server when the link is generated?

**Reply by jsjoeio** [reply]
I investigated a similar issue the other day and posted some notes [here](https://github.com/cdr/code-server/issues/2734#issuecomment-813566426). It seems like this happens when you access your code-server instance from a different region as your server.

Some questions:
- Are you using a VPN by chance?
- Where are you located and where is your digital ocean instance located?

@bpmct may have some insight as well.

**Reply by bpmct** [reply]
We have noticed connection issues related to connecting through a VPN or limited connection. Right now, the ability to capture these logs on our end is limited. @jsjoeio's suggestions are good.

**Reply by MitchHamill** [reply]
> I investigated a similar issue the other day and posted some notes [here](https://github.com/cdr/code-server/issues/2734#issuecomment-813566426). It seems like this happens when you access your code-server instance from a different region as your server.
> 
> 
> 
> Some questions:
> 
> - Are you using a VPN by chance?
> 
> - Where are you located and where is your digital ocean instance located?
> 
> 
> 
> @bpmct may have some insight as well.

Yeah this was the issue! I'm in Australia and the closes digital ocean instance was San Francisco 😬😬
Thanks so much!!

**Reply by nooobcoder** [reply]
@MitchHamill  just FYI, if you are living in Australia, the nearest endpoint for a DigitalOcean droplet should be **Bengaluru, India** and not **San Francisco** .
You can try changing the endpoint, as it should give better latency than San Francisco.

**Comment by SamHep0803** [comment, marked as answer]
It seems that the issue resolved itself.
